File: plaszczyzna/plaszczyzna_optimised.py
import random
import math
import time

MAX_ILOSC_ITERACJI = 100
srednie = []
ILOSC_PROB = 10000

def pitagoras(a, b):
    return math.hypot(a, b)  # Użyj math.hypot do obliczenia długości wektora

def avg(arr):
    return round(sum(arr) / len(arr), 2)  # Wykorzystaj funkcje wbudowane do obliczenia średniej

start_time = time.time()

for i in range(1, MAX_ILOSC_ITERACJI + 1):
    odleglosci = []

    for _ in range(ILOSC_PROB):  # Użyj pętli for zamiast while
        punkt = [random.randint(-i, i) for _ in range(2)]  # Wygeneruj punkt za pomocą listy składanej
        odleglosc = pitagoras(*punkt)  # Rozpakuj punkt jako argumenty dla funkcji pitagoras
        odleglosci.append(odleglosc)

    srednie.append(avg(odleglosci))

end_time = time.time()
elapsed_time = end_time - start_time
print("Czas wykonania:", elapsed_time)

print(srednie)

File: plaszczyzna/plaszczyzna_optimised.go
package main

import (
    "fmt"
    "math"
    "math/rand"
    "os"
    "time"
)

const (
    MAX_ILOSC_KROKOW = 1_000
    ILOSC_SYMULACJI = 1_000
)

func pitagoras(a, b int) float64 {
    return math.Sqrt(float64(a*a + b*b))
}

func avg(arr []float64) float64 {
    var sum float64
    for _, v := range arr {
        sum += v
    }
    return sum / float64(len(arr))
}

func main() {
    var srednie []float64
    punkty := make([][2]int, ILOSC_SYMULACJI)

    start := time.Now()

    for i := 1; i <= MAX_ILOSC_KROKOW; i++ {
            var suma float64
            for j := 0; j < ILOSC_SYMULACJI; j++ {
                if rand.Intn(2) == 0 {
                    punkty[j][0] += rand.Intn(2)*2 - 1
                } else {
                    punkty[j][1] += rand.Intn(2)*2 - 1
                }
                suma += pitagoras(punkty[j][0], punkty[j][1])
            }
        srednie = append(srednie, suma/float64(ILOSC_SYMULACJI))
    }   

    fmt.Printf("Czas wykonania: %s\n", time.Since(start))

    file, _ := os.Create("srednie.csv")
    for i, srednia := range srednie {
        fmt.Fprintf(file, "%d,%f\n", i+1, srednia)
    }
	file.Close()

    fmt.Println("Średnie zostały zapisane do pliku srednie.csv.")
}
